app/javascript/components/timeline-options/timeline-helper.js

```diff
@@ -0,0 +1,114 @@
+      mapsTo: 'value',
+      scaleType: 'linear',
+    },
+  },
+  legend: {
+    clickable: true,
+  },
+  tooltip: {
+    totalLabel: __('Total Events'),
+  },
+  points: {
+    radius: 3,
+    fillOpacity: 1,
+    filled: true,
+    enabled: true,
+  },
+  zoomBar: {
+    top: {
+      className: 'zoom-bar',
+      enabled: true,
+      type: 'graph_view',
+    },
+  },
+  height: '400px',
+});
+
+const smartAndOrStatements = (group, array) => {
+  let shortUrl = '';
+  if (array.length === 1) {
+    shortUrl += `&filter[]=${group}=${array[0]}`;
+  } else {
+    shortUrl += `&filter[]=${group}=[${array.toString()}]`;
+  }
+  return shortUrl;
+};
+
+/** Function to build the URL to get all events that fall under the user selected parameters. */
+export const buildUrl = (values) => {
+  // TODO: verify that everyone needs / uses the host and ems on the filters
+  let url = `/api/event_streams?limit=5000&offset=0&expand=resources&attributes=group,group_level,group_name,id,event_type,message,ems_id,type,timestamp,created_on,host,source,ems_id,ext_management_system`;
+
+  url += `&filter[]=type=${values.type}`;
+  url += smartAndOrStatements('group', values.group);
+  url += smartAndOrStatements('group_level', values.group_level);
+
+  if (values.start_date) {
+    url += `&filter[]=timestamp%3E${values.start_date[0].toISOString()}`;
+  }
+  if (values.end_date) {
+    url += `&filter[]=timestamp%3C${values.end_date[0].toISOString()}`;
+  }
+  return url;
+};
+
+/** Function to format the raw data such that it can be used by the Timeline Chart component */
+export const buildChartDataObject = (rawData) => {
+  // https://codesandbox.io/s/tender-river-9t3vu5?file=/src/index.js
+  // Link to how a dataset object should look like
+  const datasets = [];
+  rawData.resources.forEach((event) => {
+    const idx = datasets.findIndex((element) => {
+      if (element.date === event.timestamp && element.group === event.group_name) {
+        return true;
+      }
+      return false;
+    });
+    if (datasets[idx]) {
+      datasets[idx].value += 1;
+      datasets[idx].eventsObj.push(event);
+    } else { // idx comes back as -1 when its undefined
+      const obj = {
+        group: event.group_name,
+        date: event.timestamp ? event.timestamp : event.created_on,
+        value: 1,
+        eventsObj: [event],
+      };
+      datasets.push(obj);
+    }
+  });
+  return datasets;
+};
+
+/** Function to go through all the data pulled by the form submit and finds the right objects */
+export const buildDataTableObject = (pointChoice) => {
+  const tableData = [];
+  pointChoice.eventsObj.forEach((event) => {
+    const eventHost = (event.host === null) ? '' : event.host.name;
+    const eventObj = {
+      event_type: event.event_type,
+      source: event.source,
+      group_level: event.group_level,
+      provider: '', // TODO hyperlink
+      provider_username: '',
+      message: '',
+      host: eventHost, // TODO hyperlink
```

Comment:
I know its in TODO, but maybe we can avoid this variable

`host: (event.host === null) ? '' : event.host.name,`